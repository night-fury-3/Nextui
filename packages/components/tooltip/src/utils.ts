import type {TooltipPlacement} from "./types";

import {Placement} from "@react-types/overlays";

export const getOrigins = (placement: TooltipPlacement) => {
  const origins: Record<
    TooltipPlacement,
    {
      originX?: number;
      originY?: number;
    }
  > = {
    top: {
      originY: 1,
    },
    bottom: {
      originY: 0,
    },
    left: {
      originX: 1,
    },
    right: {
      originX: 0,
    },
    "top-start": {
      originX: 0,
      originY: 1,
    },
    "top-end": {
      originX: 1,
      originY: 1,
    },
    "bottom-start": {
      originX: 0,
      originY: 0,
    },
    "bottom-end": {
      originX: 1,
      originY: 0,
    },
    "right-start": {
      originX: 0,
      originY: 0,
    },
    "right-end": {
      originX: 0,
      originY: 1,
    },
    "left-start": {
      originX: 1,
      originY: 0,
    },
    "left-end": {
      originX: 1,
      originY: 1,
    },
  };

  return origins?.[placement] || {};
};

export const toReactAriaPlacement = (placement: TooltipPlacement) => {
  const mapPositions: Record<TooltipPlacement, Placement> = {
    top: "top",
    bottom: "bottom",
    left: "left",
    right: "right",
    "top-start": "top start",
    "top-end": "top end",
    "bottom-start": "bottom start",
    "bottom-end": "bottom end",
    "left-start": "left top",
    "left-end": "left bottom",
    "right-start": "right top",
    "right-end": "right bottom",
  };

  return mapPositions[placement];
};
