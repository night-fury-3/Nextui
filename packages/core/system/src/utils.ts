import type {As, RightJoinProps, PropsOf, ComponentWithAs} from "./types";

import {forwardRef as baseForwardRef} from "react";

export function forwardRef<
  Props extends object,
  Component extends As,
  CompoundComponents extends object = {},
>(
  component: React.ForwardRefRenderFunction<
    any,
    RightJoinProps<PropsOf<Component>, Props> & {
      as?: As;
    }
  >,
) {
  return baseForwardRef(component) as unknown as ComponentWithAs<Component, Props> &
    CompoundComponents;
}

export const toIterator = (obj: any) => {
  return {
    ...obj,
    [Symbol.iterator]: function () {
      const keys = Object.keys(this);
      let index = 0;

      return {
        next: () => {
          if (index >= keys.length) {
            return {done: true};
          }
          const key = keys[index];
          const value = this[key];

          index++;

          return {value: {key, value}, done: false};
        },
      };
    },
  };
};

export const mapPropsVariants = <T extends Record<string, any>, K extends keyof T>(
  props: T,
  variantKeys?: K[],
): readonly [Omit<T, K>, Pick<T, K> | {}] => {
  if (!variantKeys) {
    return [props, {}];
  }

  const omitted = Object.keys(props)
    .filter((key) => !variantKeys.includes(key as K))
    .reduce((acc, key) => ({...acc, [key]: props[key as keyof T]}), {});

  const picked = variantKeys.reduce((acc, key) => ({...acc, [key]: props[key]}), {});

  return [omitted, picked] as [Omit<T, K>, Pick<T, K>];
};
