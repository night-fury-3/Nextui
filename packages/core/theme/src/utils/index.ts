export * from "./cva";
export * from "./styles";
export * from "./types";
