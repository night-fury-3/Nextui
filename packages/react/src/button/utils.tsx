import React from 'react';
import { ButtonProps } from './button';
import { ButtonGroupConfig } from './button-group-context';

export const filterPropsWithGroup = (
  props: React.PropsWithChildren<ButtonProps>,
  config: ButtonGroupConfig
): ButtonProps => {
  if (!config.isButtonGroup) return props;
  return {
    ...props,
    auto: true,
    shadow: false,
    bordered: config.bordered ?? props.bordered,
    borderWeight: config.borderWeight ?? props.borderWeight,
    ghost: config.ghost ?? props.ghost,
    ripple: config.ripple ?? props.ripple,
    flat: config.flat ?? props.flat,
    animated: config.animated ?? props.animated,
    rounded: config.rounded ?? props.rounded,
    light: config.light ?? props.light,
    size: config.size ?? props.size,
    color: config.color ?? props.color,
    disabled: config.disabled ?? props.disabled
  };
};
