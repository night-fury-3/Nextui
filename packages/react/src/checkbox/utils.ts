import { CheckboxProps } from './checkbox';

export const mapPropsToReactAriaAttr = ({
  indeterminate,
  disabled,
  readOnly,
  required,
  checked,
  initialChecked,
  ...checkboxProps
}: CheckboxProps) => {
  return {
    ...checkboxProps,
    isIndeterminate: indeterminate,
    isDisabled: disabled,
    isReadOnly: readOnly,
    isRequired: required,
    isSelected: checked,
    defaultSelected: initialChecked
  };
};
