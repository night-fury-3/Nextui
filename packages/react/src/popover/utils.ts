import { Placement } from '@react-types/overlays';

export type PopoverPlacement =
  | 'bottom'
  | 'bottom-left'
  | 'bottom-right'
  | 'bottom-start'
  | 'bottom-end'
  | 'top'
  | 'top left'
  | 'top-right'
  | 'top-start'
  | 'top-end'
  | 'left'
  | 'left-top'
  | 'left bottom'
  | 'start'
  | 'start-top'
  | 'start-bottom'
  | 'right'
  | 'right-top'
  | 'right-bottom'
  | 'end'
  | 'end-top'
  | 'end-bottom';

export const getAriaPlacement = (placement?: PopoverPlacement) => {
  if (!placement) {
    return 'bottom' as Placement;
  }
  return placement.replace('-', ' ') as Placement;
};
