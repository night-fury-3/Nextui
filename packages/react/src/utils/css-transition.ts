import type {FC, ReactNode, RefObject} from "react";

import React, {useEffect, useState, isValidElement, cloneElement} from "react";
import {flushSync} from "react-dom";

import clsx from "./clsx";

export interface CSSTransitionProps {
  name?: string;
  visible?: boolean;
  enterTime?: number;
  leaveTime?: number;
  clearTime?: number;
  className?: string;
  onExited?: () => void;
  onEntered?: () => void;
  children?: ReactNode;
  childrenRef?: RefObject<HTMLElement>;
}

const CSSTransition: FC<CSSTransitionProps> = (props: CSSTransitionProps) => {
  const {
    children,
    onExited,
    onEntered,
    className,
    childrenRef,
    enterTime = 60,
    leaveTime = 60,
    clearTime = 60,
    visible = false,
    name = "transition",
    ...otherProps
  } = props;

  const [classes, setClasses] = useState<string>("");
  const [renderable, setRenderable] = useState<boolean>(visible);

  useEffect(() => {
    const statusClassName = visible ? "enter" : "leave";
    const time = visible ? enterTime : leaveTime;

    if (visible && !renderable) setRenderable(true);

    flushSync(() => setClasses(`${name}-${statusClassName}`));

    // set class to active
    const timer = setTimeout(() => {
      setClasses(`${name}-${statusClassName} ${name}-${statusClassName}-active`);
      if (statusClassName === "leave") {
        onExited?.();
      } else {
        onEntered?.();
      }
      clearTimeout(timer);
    }, time);

    // remove classess when animation over
    const clearClassesTimer = setTimeout(() => {
      if (!visible) {
        setClasses("");
        setRenderable(false);
      }
      clearTimeout(clearClassesTimer);
    }, time + clearTime);

    return () => {
      clearTimeout(timer);
      clearTimeout(clearClassesTimer);
    };
  }, [visible, renderable]);

  // update children ref classes
  useEffect(() => {
    if (!childrenRef?.current) return;

    const classesArr = classes.split(" ");
    const refClassesArr = childrenRef.current.className.split(" ");
    const newRefClassesArr = refClassesArr.filter((item) => !item.includes(name));

    childrenRef.current.className = clsx(newRefClassesArr, classesArr);
  }, [childrenRef, classes]);

  if (!isValidElement(children) || !renderable) return null;

  return cloneElement(children, {
    ...otherProps,
    // @ts-ignore
    className: clsx(children.props.className, className, !childrenRef?.current && classes),
  });
};

export default React.memo(CSSTransition);
