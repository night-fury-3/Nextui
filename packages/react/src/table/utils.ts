import React, { ReactNode } from 'react';
import { AriaCheckboxProps } from '@react-types/checkbox';
import { TableCollection } from '@react-types/table';

export const hasPaginationChild = (
  children: ReactNode | undefined,
  child: React.ElementType
): { hasPagination: boolean; rowsPerPage: number } => {
  const pagination = React.Children.toArray(children).find(
    (item) => React.isValidElement(item) && item.type === child
  );
  if (React.isValidElement(pagination)) {
    return {
      hasPagination: true,
      rowsPerPage: pagination?.props?.rowsPerPage || 0
    };
  }
  return { hasPagination: false, rowsPerPage: 0 };
};

export const isInfinityScroll = (collection: TableCollection<unknown>) => {
  const bodyProps = collection.body?.props;
  return (
    bodyProps?.loadingState !== undefined &&
    typeof bodyProps?.onLoadMore === 'function'
  );
};

export const mapPropsToCheckboxAttr = (ariaProps: AriaCheckboxProps) => {
  const {
    isSelected,
    defaultSelected,
    isIndeterminate,
    isReadOnly,
    isRequired,
    ...ariaCheckboxProps
  } = ariaProps;
  return {
    checked: isSelected,
    initialChecked: defaultSelected,
    indeterminate: isIndeterminate,
    readOnly: isReadOnly,
    required: isRequired,
    ...ariaCheckboxProps
  };
};
