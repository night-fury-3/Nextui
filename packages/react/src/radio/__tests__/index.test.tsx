import React from 'react';
import { mount } from 'enzyme';
import userEvent from '@testing-library/user-event';
import { nativeEvent } from '../../../tests/utils';
import Radio from '../index';
import type { ReactWrapper } from 'enzyme';

const getRadioElement = (wrapper: ReactWrapper) => {
  return wrapper.find('input');
};

const expectRadioIsChecked = (wrapper: ReactWrapper, value: boolean) => {
  expect(getRadioElement(wrapper).props().checked).toBe(value);
};

describe('Radio', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <div>
        <Radio.Group label="Options">
          <Radio value="1">Option 1</Radio>
        </Radio.Group>
        <Radio.Group label="Options" defaultValue="1" orientation="horizontal">
          <Radio value="1">1</Radio>
          <Radio value="2">2</Radio>
        </Radio.Group>
      </div>
    );
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should support circle and square', () => {
    const circle = mount(
      <Radio.Group label="Options">
        <Radio value="1">Option 1</Radio>
      </Radio.Group>
    );
    const circleDOM = circle.find('.nextui-radio-label').at(0).getDOMNode();
    expect(circleDOM.className).toContain('isSquared-false');
    expect(() => circle.unmount()).not.toThrow();
    const square = mount(
      <Radio.Group label="Options">
        <Radio value="1" isSquared>
          Option 1
        </Radio>
      </Radio.Group>
    );
    const squareDOM = square.find('.nextui-radio-label').at(0).getDOMNode();
    expect(squareDOM.className).toContain('isSquared-true');
    expect(() => square.unmount()).not.toThrow();
  });

  it('should work correctly with different sizes', () => {
    const wrapper = mount(
      <Radio.Group label="Options">
        <Radio value="xs" size="xs">
          mini
        </Radio>
        <Radio value="sm" size="sm">
          small
        </Radio>
        <Radio value="md" size="md">
          medium
        </Radio>
        <Radio value="lg" size="lg">
          large
        </Radio>
        <Radio value="xl" size="xl">
          xlarge
        </Radio>
      </Radio.Group>
    );
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should work with different colors', () => {
    const wrapper = mount(
      <Radio.Group label="Options">
        <Radio value="primary" color="primary" label="primary" />
        <Radio value="secondary" color="secondary" label="secondary" />
        <Radio value="success" color="success" label="success" />
        <Radio value="warning" color="warning" label="warning" />
        <Radio value="error" color="error" label="error" />
      </Radio.Group>
    );
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should work with different textColors', () => {
    const wrapper = mount(
      <Radio.Group label="Options">
        <Radio value="primary" labelColor="primary" label="primary" />
        <Radio value="secondary" labelColor="secondary" label="secondary" />
        <Radio value="success" labelColor="success" label="success" />
        <Radio value="warning" labelColor="warning" label="warning" />
        <Radio value="error" labelColor="error" label="error" />
      </Radio.Group>
    );
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should support react-node in description', () => {
    const wrapper = mount(
      <Radio.Group label="Options">
        <Radio value="1">Option 1</Radio>
        <Radio value="2" description="Description for Option2">
          Option 2
        </Radio>
        <Radio
          value="3"
          description={
            <p>
              <b>Description</b> for Option3
            </p>
          }
        >
          Option 3
        </Radio>
      </Radio.Group>
    );
    expect(() => wrapper.unmount()).not.toThrow();
  });

  it('should render correctly with value prop', () => {
    let value = '1';
    const wrapper = mount(
      <Radio.Group
        label="Options"
        defaultValue="2"
        onChange={(val) => (value = val)}
      >
        <Radio value="1">Option 1</Radio>
        <Radio value="2">Option 2</Radio>
      </Radio.Group>
    );
    const option1DOM = getRadioElement(wrapper)
      .at(0)
      .getDOMNode() as HTMLInputElement;
    const option2DOM = getRadioElement(wrapper)
      .at(1)
      .getDOMNode() as HTMLInputElement;
    userEvent.click(option1DOM);
    expect(value).toEqual('1');
    expect(option1DOM.checked).toEqual(true);
    expect(option2DOM.checked).toEqual(false);
    userEvent.click(option2DOM);
    expect(value).toEqual('2');
    expect(option1DOM.checked).toEqual(false);
    expect(option2DOM.checked).toEqual(true);
  });

  it('should trigger events', () => {
    let value = '';
    const changeHandler = jest.fn().mockImplementation((val) => (value = val));
    const wrapper = mount(
      <Radio.Group label="Options" onChange={changeHandler}>
        <Radio value="1">Option 1</Radio>
        <Radio value="2">Option 2</Radio>
      </Radio.Group>
    );
    wrapper
      .find('input')
      .at(0)
      .simulate('change', {
        ...nativeEvent,
        target: { checked: true }
      });
    expect(changeHandler).toHaveBeenCalled();
    expect(value).toEqual('1');
    changeHandler.mockRestore();
  });

  it('should ignore events when disabled', () => {
    const changeHandler = jest.fn();
    const wrapper = mount(
      <Radio.Group label="Options" onChange={changeHandler} isDisabled>
        <Radio value="1">Option 1</Radio>
      </Radio.Group>
    );
    wrapper
      .find('input')
      .at(0)
      .simulate('change', {
        ...nativeEvent,
        target: { checked: true }
      });
    expect(changeHandler).not.toHaveBeenCalled();
    changeHandler.mockRestore();
  });

  it('should render correctly with default-value', () => {
    const wrapper = mount(
      <Radio.Group label="Options" defaultValue="1">
        <Radio value="1">Option 1</Radio>
      </Radio.Group>
    );
    expectRadioIsChecked(wrapper, true);
  });

  it('should be warning when value unset', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    mount(
      <Radio.Group label="Options">
        <Radio>Option 1</Radio>
        <Radio value="2">Option 2</Radio>
      </Radio.Group>
    );
    expect(errorSpy).toHaveBeenCalled();
    errorSpy.mockRestore();
  });

  it('should be warning when checked is set', () => {
    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
    mount(
      <Radio.Group label="Options">
        <Radio value="1" checked>
          Option 1
        </Radio>
        <Radio value="2">Option 2</Radio>
      </Radio.Group>
    );
    expect(errorSpy).toHaveBeenCalled();
    errorSpy.mockRestore();
  });
});
