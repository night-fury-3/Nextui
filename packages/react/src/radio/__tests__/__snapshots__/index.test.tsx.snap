// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render correctly 1`] = `"<label aria-checked=\\"false\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-CNfHJ-animated-true nextui-c-cNyWQy-bAyXeE-cv\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point\\"></span>Option</span></label>"`;

exports[`Radio should support react-node in description 1`] = `"<div><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span>Option</span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span>Option 1</span><span class=\\"nextui-c-dfPuPX\\">Description for Option1</span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span>Option 1</span><span class=\\"nextui-c-dfPuPX\\"><b>Description</b> for Option1</span></label></div>"`;

exports[`Radio should work correctly with different sizes 1`] = `"<div><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-DHfpI-size-xs nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span>mini</span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-fmxghf-size-sm nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span>small</span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span>medium</span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-gedqai-size-lg nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span>large</span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-fEOUqP-size-xl nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span>xlarge</span></label></div>"`;

exports[`Radio should work with different colors 1`] = `"<div><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fIEtLE-color-secondary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-bPwnTE-color-success nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-hYvfGc-color-warning nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fmxTac-color-error nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-cyFBRX-color-default nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label></div>"`;

exports[`Radio should work with different textColors 1`] = `"<div><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-ixnjYu-color-primary nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-krgohU-color-secondary nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-byQUqd-color-success nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-gZEUWQ-color-warning nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label><label aria-checked=\\"true\\" class=\\"nextui-c-cNyWQy nextui-c-cNyWQy-fqNzuY-color-primary nextui-c-cNyWQy-cNjeeG-size-md nextui-c-cNyWQy-jMAqsJ-squared-false nextui-c-cNyWQy-cTeUqp-active-true nextui-c-cNyWQy-CNfHJ-animated-true\\"><input type=\\"radio\\" tabindex=\\"-1\\" class=\\"nextui-c-PJLV nextui-c-iFHsrU nextui-radio-input\\" value=\\"\\"><span class=\\"nextui-c-hPBoDr nextui-c-hPBoDr-XlZOi-color-error nextui-radio-name\\"><span tabindex=\\"0\\" class=\\"nextui-c-kgjvqv nextui-c-huiNHE nextui-radio-point nextui-radio-active\\"></span></span></label></div>"`;
